api: re-generate protobuf code

rolinh opened the pull request:
Commit `f18fcd9b` updated the following dependencies:

    - github.com/golang/protobuf: v1.3.2  => v1.4.2
    - google.golang.org/genproto
    - google.golang.org/grpc:     v1.26.0 => v1.27.0

However, the protobuf generated files were not re-generated which is
what this commit addresses.

The newly generated files should be backward compatible. However, the
following warning is now emitted when generating them:

```
  WARNING: Package "github.com/golang/protobuf/protoc-gen-go/generator"
           is deprecated.
           A future release of golang/protobuf will delete this package,
           which has long been excluded from the compatibility promise.
```

This is because this package was apparently never meant to be public as
explained in [this comment](https://github.com/golang/protobuf/issues/1104#issuecomment-619420855). The fix for this is to migrate to
`google.golang.org/protobuf/compiler/protogen` but this can't be done
until a release of `grpc-go` which includes [the following patch](https://github.com/grpc/grpc-go/pull/3453) is
out.

Fixes: f18fcd9b

rolinh committed:
api: bump required protoc version to 3.12.3

This is the newest version of protoc. It still produces the same
generated Go code, so the generated files are not affected.

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

gandro reviewed (approved):
Thanks. Looks good to me, modulo the issue in the lints (looks like a vendoring issue with the generated code).

The warning is unfortunate, but we can safely ignore it, as long as we don't accidentally bump the dependency. The old protobuf generator might be deprecated, but the new version does not come with built-in grpc support (which we need), and the standalone successor of the grpc plugin is not released yet.

I wonder if we should add a notice message in the Makefile to tell users to ignore the warning?

rolinh committed:
api: re-generate protobuf code

Commit `f18fcd9b` updated the following dependencies:

  - github.com/golang/protobuf: v1.3.2  => v1.4.2
  - google.golang.org/genproto
  - google.golang.org/grpc:     v1.26.0 => v1.27.0

However, the protobuf generated files were not re-generated which is
what this commit addresses.

The newly generated files should be backward compatible. However, the
following warning is now emitted when generating them:

  WARNING: Package "github.com/golang/protobuf/protoc-gen-go/generator"
           is deprecated.
           A future release of golang/protobuf will delete this package,
           which has long been excluded from the compatibility promise.

This is because this package was apparently never meant to be public as
explained in this comment[0]. The fix for this is to migrate to
`google.golang.org/protobuf/compiler/protogen` but this can't be done
until a release of `grpc-go` which includes the following patch[1] is
out. Update the Makefile to output a note with regard to this when generating
the files

It appears that with the newly generated files, protobuf now needs to
be marked as an explicit dep. Fix it by by running:

    go mod tidy && go mod vendor && go mod verify

Fix: f18fcd9b

[0]: https://github.com/golang/protobuf/issues/1104#issuecomment-619420855
[1]: https://github.com/grpc/grpc-go/pull/3453

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>
